src/deadline/job_attachments/api/manifest.py

```diff
@@ -247,3 +260,184 @@ def process_output(status: FileStatus, path: str, output_diff: ManifestDiff):
             process_output(fast_diff_item[1], fast_diff_item[0], output)
 
     return output
+
+
+def _manifest_upload(
+    manifest_file: str,
+    s3_bucket_name: str,
+    s3_key_prefix: str,
+    s3_metadata: Dict[str, Any],
+    boto_session: boto3.Session,
+    logger: ClickLogger = ClickLogger(False),
+):
+    """
+    BETA API - This API is still evolving but will be made public in the near future.
+    API to upload a job attachment manifest to the Content Addressable Storage. Manifests will be
+    uploaded to s3://{s3_bucket_name}/{root_prefix}/Manifest/{s3_key_prefix} as per the Deadline CAS folder structure.
```

Comment:
From 
https://github.com/aws-deadline/deadline-cloud/blob/b307b21e6a4cd04025d1ead8d8f232114be06983/src/deadline/client/cli/_groups/manifest_group.py#L316-L317

and 
https://github.com/aws-deadline/deadline-cloud/blob/b307b21e6a4cd04025d1ead8d8f232114be06983/src/deadline/client/cli/_groups/manifest_group.py#L333-334

s3_key_bucket name is bucket name, s3_key_prefix is root prefix, where is this s3_key_prefix from? Did I miss anything?

Reply:
This is related to my confusion in an earlier comment 

https://github.com/aws-deadline/deadline-cloud/pull/471/files#r1809364159

Reply:
Ah - ok this is a documentation error. I will change this.

Reply:
OK now i remember - I was in the middle of adding this,  `s3://{s3_bucket_name}/{cas_prefix}/Manifest/{s3_key_prefix}/{manifest_file_name}`

The CAS `Manifest` folder can have subfolder paths, eg job/step/task that we use today.